manifests/daemon_reload.pp

```diff
@@ -7,15 +7,55 @@
 #
 # @param enable
 #   Enable the reload exec
-#
 #   * Added in case users want to disable the reload globally using a resource collector
 #
+# @param user
+#   Specify user name of `systemd --user` to reload. This not supported **below** Redhat 9,
+#   Ubuntu 22.04 or Debian 12.
+#
+# @example Force reload the system systemd
+#   notify{ 'fake event to notify from':
+#     notify => Systemd::Daemon_reload['special']
+#   }
+#   systemd::daemon_reload{ 'special': }
+#
+# @example Force reload a systemd --user
+#   notify{ 'fake event to notify from':
+#     notify => Systemd::Daemon_reload['steve_user']
+#   }
+#   systemd::daemon_reload{ 'steve_user':
+#     user => 'steve',
+#   }
+#
 define systemd::daemon_reload (
   Boolean $enable = true,
+  Optional[String[1]] $user = undef,
 ) {
+  include systemd
+
   if $enable {
-    exec { "${module_name}-${name}-systemctl-daemon-reload":
-      command     => 'systemctl daemon-reload',
+    if $user {
+      if ($facts['os']['family'] == 'RedHat' and versioncmp($facts['os']['release']['major'],'9') < 0 ) or
+      ($facts['os']['name'] == 'Debian' and versioncmp($facts['os']['release']['major'],'12') < 0 ) or
+      ($facts['os']['name'] == 'Ubuntu' and versioncmp($facts['os']['release']['major'],'22.04') < 0 ) {
+        fail('systemd::daemon_reload_for a user is not supported below RedHat 9, Debian 12 or Ubuntu 22.04')
+      }
+
+      $_title   = "${module_name}-${name}-systemctl-user-${user}-daemon-reload"
+      # Must use full path to systemctl here.
+      $_command = ['machinectl', 'shell' , "${user}@.host", '/usr/bin/systemctl', '--user', 'daemon-reload']
+
+      if $systemd::machinectl_package {
```

Comment:
does it make sense to have an explicit parameter that configures the package installation? People might manage it already on other places in the catalog. That would also allow us to remove the `stdlib::ensure_packages()` call wich isn't 100% reliable anyways.

Reply:
I considered it for sure - I decided as it had no apparent configuration you could make for the package it was not needed. Your point is valid that ensure_packages can go wrong. So a 

* `systemd::manage_containers`
* `systemd::containers_package`

and a new class systemd::containers ?

? since that's the package name.

Prefer that?

Reply:
I think that's a good way forward.